### PR TITLE
Don't handle bias inside cudnn_convolution*

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Compared to cuDNN bias, PyTorch add has the following advantage:
- faster, especially for backward (see: https://github.com/zasdfgbnm/things/blob/master/2019/conv-backward-profile.md)
- handles 64bit indexing automatically
- has less code, less maintenance effort

@ngimel I submit this PR early so the CI could start building it. But I have not tested it locally yet (still waiting for compiling).

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
@houseroad Should we worry about the backward compatibility on the signature of `cudnn_convolution*`? IMHO, I think there is nothing to worry about because it is not part of the public API, so no JIT model should have it(otherwise how can a JIT model run `cudnn_convolution` on CPU?).

File: tools/autograd/derivatives.yaml [hunk 1]
```diff
@@ -1407,17 +1407,17 @@
 - name: _cudnn_ctc_loss(Tensor log_probs, Tensor targets, int[] input_lengths, int[] target_lengths, int blank, bool deterministic, bool zero_infinity) -> (Tensor, Tensor)
   log_probs: _cudnn_ctc_loss_backward(grad, result0, result1, zero_infinity)
 
-- name: cudnn_convolution_transpose(Tensor self, Tensor weight, Tensor? bias, int[] padding, int[] output_padding, int[] stride, int[] dilation, int groups, bool benchmark, bool deterministic) -> Tensor
-  self, weight, bias: cudnn_convolution_transpose_backward(self, grad, weight, padding, output_padding, stride, dilation, groups, benchmark, deterministic, grad_input_mask)
+- name: cudnn_convolution_transpose(Tensor self, Tensor weight, int[] padding, int[] output_padding, int[] stride, int[] dilation, int groups, bool benchmark, bool deterministic) -> Tensor
```
Comment (houseroad) [inline tools/autograd/derivatives.yaml]: Yes, these are breaking changes. We have two options to address the problem:
1) Make the API still backward compatible,
2) Use the overload name to deprecate the old functions, make the old one like cudnn_convolution_transpose.deprecated,
and new one cudnn_convolution_transpose.

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit ebb445b0:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=ebb445b0a7b01379360326b38876a160101e4794).

### :detective: 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4038187) pytorch_windows_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=78486625) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=78486625))
<details>
<summary>
<code>CMake Error at torch_cuda_generated_THCTensorMathReduce.cu.obj.Release.cmake:281 (message):</code>
</summary>

```
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/system/cuda/detail/core/util.h(170): error C2923: 'std::_Select<__formal>::_Apply': 'ScanOp' is not a valid template type argument for parameter '<unnamed-symbol>'
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/system/cuda/detail/core/util.h(170): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/system/cuda/detail/core/util.h(170): error C2144: syntax error: 'unknown-type' should be preceded by ')'
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/system/cuda/detail/core/util.h(170): error C2144: syntax error: 'unknown-type' should be preceded by ';'
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/system/cuda/detail/core/util.h(170): error C2238: unexpected token(s) preceding ';'
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/system/cuda/detail/core/util.h(170): error C2059: syntax error: ')'
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/system/cuda/detail/core/util.h(170): error C2988: unrecognizable template declaration/definition
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/system/cuda/detail/core/util.h(170): error C2059: syntax error: '<end Parse>'
-- Removing C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/./torch_cuda_generated_THCTensorMathReduce.cu.obj
"C:/Program Files/CMake/bin/cmake.exe" -E remove C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/./torch_cuda_generated_THCTensorMathReduce.cu.obj
CMake Error at torch_cuda_generated_THCTensorMathReduce.cu.obj.Release.cmake:281 (message):
  Error generating file
  C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/./torch_cuda_generated_THCTensorMathReduce.cu.obj


L=ON -D build_configuration:STRING=Release -D generated_file:STRING=C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/./torch_cuda_generated_THCTensorCopy.cu.obj -D generated_cubin_file:STRING=C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/./torch_cuda_generated_THCTensorCopy.cu.obj.cubin.txt -P C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/torch_cuda_generated_THCTensorCopy.cu.obj.Release.cmake" 
-- Removing C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/./torch_cuda_generated_THCTensorCopy.cu.obj
"C:/Program Files/CMake/bin/cmake.exe" -E remove C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/./torch_cuda_generated_THCTensorCopy.cu.obj
-- Generating dependency file: C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/torch_cuda_generated_THCTensorCopy.cu.obj.NVCC-depend
C:/Users/circleci/project/aten/src/THCUNN -IC:/Users/circleci/project/aten/src/ATen/cuda -IC:/Users/circleci/project/build/caffe2/aten/src -IC:/Users/circleci/project/aten/../third_party/catch/single_include -IC:/Users/circleci/project/aten/src/ATen/.. -IC:/Users/circleci/project/build/caffe2/aten/src/ATen -IC:/Users/circleci/project/c10/cuda/../.. -IC:/Users/circleci/project/c10/../ -IC:/Users/circleci/project/caffe2/../torch/csrc/api -IC:/Users/circleci/project/caffe2/../torch/csrc/api/include
THCTensorCopy.cu 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

**Comment by zasdfgbnm** [comment]
cc: @csarofeen @ptrblck for visibility

**Comment by ngimel** [comment]
@houseroad are you ok with the overloads here? Note, `cudnn_convolution_backward_bias` is removed altogether (good riddance), and someone calling deprecated overloads won't be able to backward through them. This all is only backward-related though, so should not impact serialized models for inference.

**Review by houseroad — COMMENTED** [review]
BC part LGTM. Btw, some conflict happens, need to rebase to master.

**Pull request closed without merging**